timber-lint/src/main/java/timber/lint/WrongTimberUsageDetector.kt

```diff
@@ -0,0 +1,808 @@
+import org.jetbrains.uast.UIfExpression
+import com.intellij.psi.PsiMethodCallExpression
+import com.intellij.psi.PsiLiteralExpression
+import com.intellij.psi.PsiType
+import com.intellij.psi.PsiClassType
+import com.android.tools.lint.checks.StringFormatDetector
+import com.android.tools.lint.client.api.TYPE_BOOLEAN
+import com.android.tools.lint.client.api.TYPE_BOOLEAN_WRAPPER
+import com.android.tools.lint.client.api.TYPE_BYTE
+import com.android.tools.lint.client.api.TYPE_BYTE_WRAPPER
+import com.android.tools.lint.client.api.TYPE_CHAR
+import com.android.tools.lint.client.api.TYPE_DOUBLE
+import com.android.tools.lint.client.api.TYPE_DOUBLE_WRAPPER
+import com.android.tools.lint.client.api.TYPE_FLOAT
+import com.android.tools.lint.client.api.TYPE_FLOAT_WRAPPER
+import com.android.tools.lint.client.api.TYPE_INT
+import com.android.tools.lint.client.api.TYPE_INTEGER_WRAPPER
+import com.android.tools.lint.client.api.TYPE_LONG
+import com.android.tools.lint.client.api.TYPE_LONG_WRAPPER
+import com.android.tools.lint.client.api.TYPE_NULL
+import com.android.tools.lint.client.api.TYPE_OBJECT
+import com.android.tools.lint.client.api.TYPE_SHORT
+import com.android.tools.lint.client.api.TYPE_SHORT_WRAPPER
+import com.android.tools.lint.client.api.TYPE_STRING
+import com.android.tools.lint.detector.api.Category.Companion.CORRECTNESS
+import com.android.tools.lint.detector.api.Category.Companion.MESSAGES
+import com.android.tools.lint.detector.api.ConstantEvaluator.evaluateString
+import com.android.tools.lint.detector.api.Detector.UastScanner
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.Scope.Companion.JAVA_FILE_SCOPE
+import com.android.tools.lint.detector.api.Severity.ERROR
+import com.android.tools.lint.detector.api.Severity.WARNING
+import org.jetbrains.uast.ULiteralExpression
+import org.jetbrains.uast.USimpleNameReferenceExpression
+import com.intellij.psi.PsiField
+import com.intellij.psi.PsiParameter
+import java.lang.Byte
+import java.lang.Double
+import java.lang.Float
+import java.lang.IllegalStateException
+import java.lang.Long
+import java.lang.Short
+import java.util.Calendar
+import java.util.Date
+import java.util.regex.Pattern
+
+class WrongTimberUsageDetector : Detector(), UastScanner {
+  override fun getApplicableMethodNames() = listOf("tag", "format", "v", "d", "i", "w", "e", "wtf")
+
+  override fun visitMethodCall(context: JavaContext, node: UCallExpression, method: PsiMethod) {
+    val methodName = node.methodName
+    val evaluator = context.evaluator
+
+    if ("format" == methodName && evaluator.isMemberInClass(method, "java.lang.String")) {
+      checkNestedStringFormat(context, node)
+      return
+    }
+    // As of API 26, Log tags are no longer limited to 23 chars.
+    if ("tag" == methodName
+      && evaluator.isMemberInClass(method, "timber.log.Timber")
+      && context.project.minSdk <= 25
```

Comment:
```suggestion
      && context.project.minSdk < 26
```
Don't like the disparity with the comment